### DISCUSSION TITLE
[BUG] - blank page - docker

### DISCUSSION THREAD

**Discussion by jenaye** [poowf/invoiceneko#199] [Q&A]
Hi, i tried to install the tool and i got blank page and this stack-trace, can you fix it ? 

![image](https://user-images.githubusercontent.com/15458329/136122722-3ff829d1-3d20-40bc-a2a9-fd337b21293a.png)

![image](https://user-images.githubusercontent.com/15458329/136123066-c7bd6c54-2e5b-40a3-a11e-888e522765d9.png)

Also, can be nice to add some screenshot of the application or maybe demo idk

Peace
Jenaye,

**Comment by jenaye** [comment]
Lil update, my bad, 

I forgot to copy the .env file

but I have another stacktrace, i had to `sudo chmod -R ugo+rw storage` and do the same for the folder bootstrap

and we have 

![image](https://user-images.githubusercontent.com/15458329/136124129-2722a72a-5e77-4323-8dba-25cc48bc1010.png)


i took ure in fake .env to test skip this step and now i got this error : 

![image](https://user-images.githubusercontent.com/15458329/136124422-3c8a034d-c17f-423b-82b1-e5c49551e318.png)

**Comment by zanechua** [comment]
Hey @jenaye,

What exactly are you using?

`docker run` ?

The `docker-compose` is an environment image at the moment, not really meant for production usage.

Please see:

https://github.com/poowf/invoiceneko/blob/develop/resources/docs/1.0/development.md


This is not really a bug, would be good if we can move this to discussions.

**Comment by jenaye** [comment]
Hey @zanechua 
thank you for your reactivity! 
actually i wanted to test the tool quickly, so i just run the docker-compose up command 

I'll follow your guide and come back to you soon

**Comment by jenaye** [comment]
ok so its look working but i had to install  yarn and add some permissions to boostrap and storage folder

![image](https://user-images.githubusercontent.com/15458329/136128383-ea06a650-288b-4eac-a125-1c040d05dfee.png)


now when i'm trying to create account i got this error : 

![image](https://user-images.githubusercontent.com/15458329/136128506-41388895-275a-4e8c-bbe0-45c52209c092.png)

**Reply by zanechua** [reply]
That's because the SMTP credentials are not configured.

You can configure it to e.g. mailtrap so that you can get the sign up email.

I should probably have handled this in a better way.

**Comment by jenaye** [comment]
Hey @zanechua ,
Is it possible to create a user without validating his email etc? 

The tool looks cool but I'm disappointed because we're talking about people who need to manage their invoices  and clients as freelancers for example, we don't need to have validation by email 

In my case, I wanted to host it on a promox cut off from the world, and I would access the application through a ssh -L in my local network

By the way, it would be a pleasure to do a pentest of the application but for that I would have to launch it and I don't want to set up another service by myself or create account somewhere

Don't take it the wrong way, I would just like to be able to launch the tool in 1 command because it looks interesting

**Reply by zanechua** [reply]
The code is open-sourced, you're free to make any changes to it per the license, I may update the handling of email events but not at this time. I'm working towards a refactor and this will be done afterwards.

I think the current deployment process is pretty straightforward? If I'm free I can possibly make a compose environment for a production based setup but you're not gonna get away with "one-click" setup if you have to modify the `.env` in any way.

Once the refactor is done, I'll have more time to circle back and make deployment updates. Otherwise no promises at the moment.
